### PR TITLE
Awood/clowder config

### PR DESCRIPTION
**Pull request by awood**
Add a ClowderJsonFileSource that will allow us to load the Clowder JSON file and provide [JSON Pointer](https://www.baeldung.com/json-pointer) expressions that allow us to select values from the JSON.

For example, if we need to get the hostname from 

```json
"kafka": {
    "brokers": [{
        "hostname": "env-rhsm-kafka.rhsm.svc",
        "port": 29092
    },
    {
        "hostname": "other-env-rhsm-kafka.rhsm.svc",
        "port": 29092
    }
]}
```
We can use the JSON pointer expression `/kafka/brokers/0/hostname` to get the hostname of the first element.

JSON Pointer is not perfect for our usage but it is what Jackson supports natively.  For example, in the JSON above, suppose we wanted both hostnames.  The `jq` utility can use a filter like `.kafka.brokers[].hostname` to fetch all the hostnames.  JSON Pointer is not that expressive.  Instead we would have to use the expression `/kafka/brokers` and then deserialize each element to a `BrokerData` object which had `hostname` and `port` fields.  This arrangement in not ideal, but I think it is one we can live with.

### CODE REVIEW

**Review by kahowell — CHANGES_REQUESTED** [review]
I see where you were going with this. I don't like the idea of needing to handle JsonPointer all over the place though...

I've been noodling this some bit... There's a different approach I'd like your thoughts on. I want to decompose using clowder config into three steps.

1. Parse the clowder config into something that we can work with in-memory
2. Map the clowder config object into properties that can be injected into the spring environment. (would need to maintain a mapping of something like JsonPath or JsonPointer to property name). Note that if we map to all upper-case names consistent with the env vars, that works. This mapping could be either data, or just logic. I was looking at the endpoints array, and unfortunately, don't think we can integrate that schema without at least JsonPath, at which point, I'm happy to simply use code on a JsonNode.
3. Inject the mapped properties into the Spring Environment.

I looked into this a bit, and we can accomplish (3) by implementing `org.springframework.boot.env.EnvironmentPostProcessor` and registering it in `src/main/resources/META-INF/spring.factories`

Thoughts?

File: src/main/java/org/candlepin/subscriptions/clowder/ClowderJson.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+/**
+ * Accessor class to pull values from the Clowder JSON file. Values are queried using the <a
+ * href="http://tools.ietf.org/html/draft-ietf-appsawg-json-pointer-03">JSON Pointer</a>
+ * specification.
+ *
+ * <p>Jackson has the, in my opinion, odd behavior of returning a <i>default</i> value if the JSON
+ * Pointer can't be resolved into the requested data type. This class disagrees with that decision
+ * and instead throws exceptions if the data can not be converted to the requested type.
+ */
+public class ClowderJson {
```
Comment (kahowell) [inline src/main/java/org/candlepin/subscriptions/clowder/ClowderJson.java:38]: Would like to see these classes land in `swatch-core`.

**Review by kahowell — CHANGES_REQUESTED** [review]

File: src/main/java/org/candlepin/subscriptions/ApplicationProperties.java [hunk 2]
```diff
@@ -97,6 +101,9 @@
    */
   private Duration productProfileListCacheTtl = Duration.ofMinutes(5);
 
+  /** Amount of time to cache the Clowder JSON before re-reading from the filesystem */
+  private Duration clowderJsonCacheTtl = Duration.ofMinutes(5);
+
```
[the remark this answers is not in the document]
Comment (awood) [reply src/main/java/org/candlepin/subscriptions/ApplicationProperties.java]: So the caching is an artifact of descending from `StructuredFileSource`.  I mean, I can just hardcode the constructor to use a `Duration` of a decade or something or refactor `StructuredFileSource` to allow for sources that never invalidate.  When I wrote this bit, I figured we'd just leave it as a property but never actually bother setting it in a real properties file.  I can also just keep the property but set the default value to a decade.  Let me know what you think.

[the remark this answers is not in the document]
Comment (kahowell) [reply src/main/java/org/candlepin/subscriptions/ApplicationProperties.java]: A decade is fine by me, just leave a note about it, please.

**Pull request closed without merging**